src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Android.App;
+using Android.OS;
+using Android.Runtime;
+using Microsoft.DotNet.XHarness.TestRunners.Common;
+using Microsoft.DotNet.XHarness.TestRunners.Xunit;
+
+// parsing of arguments for which test methods to run
+// properly set the return-code
+// to do the filter and excludes
+// automatically return the results file
+
+namespace Microsoft.DotNet.XHarness.InstrumentationBase.Xunit
+{
+    public abstract class AndroidInstrumentationBase : Instrumentation
+    {
+        private const string ResultsFileArgumentName = "results-file-name";
+        private const string ResultsPathArgumentName = "test-results-path";
+        private const string SummaryMessageArgumentName = "test-execution-summary";
+        private const string ExitCodeArgumentName = "return-code";
+        private const string ExcludeCategoriesArgumentName = "exclude-categories";
+        private const string ExcludeMethodArgumentName = "exclude-method";
+        private const string IncludeMethodArgumentName = "include-method";
+        private const string ExcludeClassArgumentName = "exclude-class";
+        private const string IncludeClassArgumentName = "include-class";
+
+        protected Dictionary<string, string> bundleArguments;
+
+        protected AndroidInstrumentationBase()
+        {
+        }
+        protected AndroidInstrumentationBase(IntPtr handle, JniHandleOwnership transfer)
+            : base(handle, transfer)
+        {
+        }
+
+        protected virtual IEnumerable<string> Tests
+        {
+            get
+            {
+                yield return Assembly.GetExecutingAssembly().Location;
+            }
+        }
+
+        public override void OnCreate(Bundle arguments)
+        {
+            base.OnCreate(arguments);
+
+            foreach (var key in arguments.KeySet())
+            {
+                string value = arguments.GetString(key);
+                if (!string.IsNullOrEmpty(value))
+                {
+                    bundleArguments.Add(key, value);
+                }
+            }
+
+            if (bundleArguments.ContainsKey(ResultsFileArgumentName) == false)
```

Comment:
This `== false` seems fishy

Reply:
You could also do `bundleArguments.TryGetValue(key, out var value)` and just let the system do all the checks for you.